packages/flutter_tools/lib/src/android/setup_split_aot.dart

```diff
@@ -0,0 +1,482 @@
+      print('FINDING MODULES ${file.path}');
+      for (Map loadingUnitMetadata in manifest['loadingUnits']) {
+        if (loadingUnitMetadata['id'] == 1) continue;
+        loadingUnits.add(LoadingUnit(
+          id: loadingUnitMetadata['id'],
+          path: loadingUnitMetadata['path'],
+          libraries: List<String>.from(loadingUnitMetadata['libraries']))
+        );//'module${loadingUnitMetadata['id'].toString()}');
+      }
+      break;
+    }
+    files.removeLast();
+  }
+
+  _compareAgainstGolden(loadingUnits, androidDir, setGolden: setGolden);
+
+  return loadingUnits;
+}
+
+bool _compareAgainstGolden(List<LoadingUnit> loadingUnits, Directory androidDir, {bool setGolden = true}) {
+  File goldenFile = androidDir.childFile('loading_unit_golden.json');
+
+  if (!goldenFile.existsSync()) {
+    print('No loading_unit_golden.json found');
+    if (setGolden) {
+      print('Setting this run\'s output as initial golden');
+      _createLoadingUnitGolden(loadingUnits, goldenFile);
+      return true;
+    } else {
+      print('Run split AOT setup to generate a golden.');
+      return false;
+    }
+  }
+
+  var goldenManifest = jsonDecode(goldenFile.readAsStringSync());
+  List<LoadingUnit> missingLoadingUnits = [];
+  List<LoadingUnit> generatedLoadingUnits = List<LoadingUnit>.from(loadingUnits);
+  Map manifest = jsonDecode(goldenFile.readAsStringSync());
+
+  for (Map loadingUnitMetadata in goldenManifest['loadingUnits']) {
+    if (loadingUnitMetadata['id'] == 1) continue;
+    LoadingUnit goldenUnit = LoadingUnit(
+      id: loadingUnitMetadata['id'],
+      path: loadingUnitMetadata['path'],
+      libraries: List<String>.from(loadingUnitMetadata['libraries'])
+    );
+    bool found = false;
+    for (LoadingUnit generatedUnit in loadingUnits) {
+      if (goldenUnit.hashCode == generatedUnit.hashCode) {
+        found = true;
+        generatedLoadingUnits.remove(generatedUnit);
+        break;
+      }
+    }
+    if (!found) {
+      missingLoadingUnits.add(goldenUnit);
+    }
+  }
+  if (generatedLoadingUnits.isNotEmpty) {
+    print('New loading units were found: $generatedLoadingUnits');
+  }
+  if (missingLoadingUnits.isNotEmpty) {
+    print('Previously existing loading units no longer exist: $missingLoadingUnits');
+  }
+  if (missingLoadingUnits.isEmpty && generatedLoadingUnits.isEmpty) {
+    print('No change in generated loading units.');
+  } else {
+    // verify with user if they want to set this configuration as new golden.
+  }
+  _createLoadingUnitGolden(loadingUnits, goldenFile);
+}
+
+void _createLoadingUnitGolden(List<LoadingUnit> loadingUnits, File goldenFile) {
+  goldenFile.createSync(recursive: true);
+
+//   String goldenContents = ''
+// '''
+// # The contents of this file are generated via the prebuild step of
+// # Flutter split AOT bundle building. It is not recommended to edit
+// # this manually.
+// #
+// # This file contains a list of loading units/dylibs that were
```

Comment:
dylib is really a macOS/iOS specific format, these should be referred to as shared libraries.